shell/platform/fuchsia/flutter/session_connection.cc

```diff
@@ -92,6 +128,12 @@ void SessionConnection::EnqueueClearOps() {
 
 void SessionConnection::PresentSession() {
   TRACE_EVENT0("gfx", "SessionConnection::PresentSession");
+
+  if (frames_in_flight_allowed_ == 0)
+    return;
```

Comment:
I see. It just worries me that someone could change something later and this would cause the loop to stall out and be hard to track down. Add comment about where the loop should pick back up if this happens.

Reply:
Done

Reply:
I don't see any changes. Forgot to upload?

Reply:
I meant to save that comment and then only submit it with the new patch, but I guess that's only when I "Start a review".

Reply:
Should this set present present_session_pending_ = true? Otherwise I don't think this is guaranteed to not stall.

Reply:
Yes, but `present_session_pending_` will already always be true since we only set it to false after the if guard evaluates to false.

Reply:
You don't check the input though, so add an assert to see that it's true inside the if statement. Then LGTM from me.

Reply:
Good point, done in upcoming CL